src/test/java/healthy/lifestyle/backend/admin/users/repository/UserAdminRepositoryTest.java

```diff
@@ -0,0 +1,126 @@
+            List<Integer> resultSeeds) {
+
+        // Given
+        Role roleUser = dataHelper.createRole("ROLE_USER");
+        Role roleAdmin = dataHelper.createRole("ROLE_ADMIN");
+
+        Country country1 = dataHelper.createCountry(1);
+        Country country2 = dataHelper.createCountry(2);
+
+        User user1 = dataHelper.createUser("1", roleUser, country1, null, 34);
+        User user2 = dataHelper.createUser("2", roleUser, country2, null, 16);
+        User admin1 = dataHelper.createUser("3", roleAdmin, country2, null, 45);
+        User admin2 = dataHelper.createUser("4", roleAdmin, country2, null, 21);
+
+        Optional<Role> roleFilter = Stream.of(roleUser, roleAdmin)
+                .filter(role -> role.getName().equals(roleName))
+                .findFirst();
+        Optional<Country> countryFilter = Stream.of(country1, country2)
+                .filter(country -> country.getName().equals(countryName))
+                .findFirst();
+
+        // When
+        Optional<List<User>> resultOptional = userAdminRepository.findByFilters(
+                roleFilter.orElse(null), username, email, fullName, countryFilter.orElse(null), age);
+
+        // Then
+        assertTrue(resultOptional.isPresent());
+        List<User> result = resultOptional.get();
+
+        assertEquals(resultSeeds.size(), result.size());
+        for (int i = 0; i < resultSeeds.size(); i++) {
+            assertEquals("username-" + resultSeeds.get(i), result.get(i).getUsername());
+            assertEquals(
+                    "username-" + resultSeeds.get(i) + "@email.com",
+                    result.get(i).getEmail());
+            assertEquals("Full Name " + resultSeeds.get(i), result.get(i).getFullName());
+        }
+    }
+
+    static Stream<Arguments> multipleValidFilters() {
+        return Stream.of(
+                Arguments.of(null, null, null, null, null, null, List.of(1, 2, 3, 4)),
+                Arguments.of(
+                        "ROLE_USER", "NameUser", "FullNameUser", "EmailUser", "Country 3", 78, Collections.emptyList()),
+                Arguments.of("ROLE_USER", null, null, null, null, null, List.of(1, 2)),
+                Arguments.of("ROLE_ADMIN", null, null, null, null, null, List.of(3, 4)),
+                Arguments.of(
+                        "ROLE_USER", "username-1", "username-1@email.com", "Full Name 1", "Country 1", 34, List.of(1)),
+                Arguments.of(null, null, null, null, "Country 2", null, List.of(2, 3, 4)),
+                Arguments.of(null, null, null, null, "Country 1", null, List.of(1)),
+                Arguments.of("ROLE_USER", null, "username-2@email.com", null, null, null, List.of(2)),
+                Arguments.of("ROLE_ADMIN", null, null, "Full Name 3", null, null, List.of(3)),
+                Arguments.of("ROLE_USER", null, null, null, "Country 2", null, List.of(2)),
+                Arguments.of("ROLE_ADMIN", null, null, null, null, 45, List.of(3)));
+    }
```

Comment:
Please add some parameters:

all users of all roles - done
empty list - done

role user and username - done
role user and email - todo
role user and full name - todo
role user and country - done
role user and age - todo
role user and all fields - done
all users with role user - done

role admin and username - todo
role admin and email - todo
role admin and full name - done
role admin and country - todo
role admin and age - done
role admin and all fields - todo
all users with role admin - done

no role and username - todo
no role and email - todo
no role and full name - todo
no role and country - done
no role and age - todo